src/game/objects/King/KingFall.js

```diff
@@ -0,0 +1,45 @@
+import { AssetsManager } from '../../assets';
+import { KingAnimated } from './KingAnimated';
+
+const SPRITE_ANIMATION_UPDATE = 10;
+
+class KingFall extends KingAnimated {
+  /**
+   * @param {import('./King').King} king
+   */
+  constructor(king) {
+    super(king, [AssetsManager.kingFall], SPRITE_ANIMATION_UPDATE);
+  }
+
+  /**
+   * @param {import('../../shared').Frame} frame
+   */
+  update(frame) {
+    if (this.king.isOnGround) {
+      return this.king.transitionState('ground').onUpdate(frame);
+    }
+    if (this.king.keyboard.pressed('ArrowRight')) {
+      this.king.rigidbody.velocity = this.king.rigidbody.velocity.setX(1);
+    }
+    if (this.king.keyboard.pressed('ArrowLeft')) {
+      this.king.rigidbody.velocity = this.king.rigidbody.velocity.setX(-1);
+    }
+    if (!this.king.keyboard.pressed('ArrowRight') && !this.king.keyboard.pressed('ArrowLeft')) {
+      this.king.rigidbody.velocity = this.king.rigidbody.velocity.setX(0);
+    }
+    if (this.king.keyboard.pressed('x')) {
+      if (this.king.canAttack) {
+        if (this.king.canAttack) {
```

Comment:
Chyba jest zduplikowany `if`? A przynajmniej nie widzę możliwości w której wejdziemy do zewnętrznego `if`a, a do wewnętrznego nie. To samo pytanie do `KingJump`, `KingRunLeft` i `KingRunRight`.

Reply:
Oczywiście, że jest zduplikowany, nie wiem jak mogłam tego nie zauważyć 🤦‍♀️. Tak to jest jak się na szybko robi kopiuj-wklej. Zaraz poprawiam